src/main/java/com/Elessar/app/server/LogOnHandler.java

```diff
@@ -36,24 +45,30 @@ public void handle(HttpExchange he) throws IOException {
             final LogonRequest logonRequest = LogonRequest.parseFrom(is);
             final LogonResponse.Builder logonResponse = LogonResponse.newBuilder();
             final String userName = logonRequest.getName();
-            final String password = logonRequest.getPassword();
-            if (!userData.containsKey(userName)) {
-                logonResponse.setSuccess(false).setFailReason("User Name " + userName + " is NOT Registered !");
-                he.sendResponseHeaders(400, 0);
-            } else if (!userData.get(userName).getPassword().equals(password)){
-                //System.out.printf("user = %s, password1 = %s, password2 = %s\n", userName, userData.get(userName).getPassword(), password);
-                logonResponse.setSuccess(false).setFailReason("Incorrect Password !");
-                he.sendResponseHeaders(400, 0);
-            } else if (userData.get(userName).getOnlineStatus()) {
-                logonResponse.setSuccess(false).setFailReason(userName + " is already log on !");
+            final Map<String, String> filters = new HashMap<>();
+            filters.put("name", logonRequest.getName());
+            filters.put("password", logonRequest.getPassword());
+
+            Iterator<Document> cursor = users.find(filters).iterator();
+            if (!cursor.hasNext()) {
+                logger.info("User {} and password combination does NOT exist !", userName);
+                logonResponse.setSuccess(false).setFailReason("User " + userName + " password combination does NOT exist !");
                 he.sendResponseHeaders(400, 0);
             } else {
-                userData.get(userName).setOnline();
-                logonResponse.setSuccess(true);
-                he.sendResponseHeaders(200, 0);
+                Document doc = cursor.next();
+                if (users.isFieldEqual(doc, "online", "true")) {
```

Comment:
don't think it's helpful to have method such as isFieldEqual().
you can just operate on "doc" directly